bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WindowsDefenderConfigurator.java

```diff
@@ -150,8 +153,8 @@ private static Boolean runExclusionCheck(IProgressMonitor m, Optional<Path> inst
 		if (!isWindowsDefenderActive(monitor.split(1))) {
 			return Boolean.FALSE;
 		}
-
-		HandlingOption decision = askForDefenderHandlingDecision();
+		Display display = Display.getDefault();
```

Comment:
Should this not use `getDisplay()`?

Reply:
You mean the one from before?
I have have replaced that now with just this single call to `Display.getDefault()`. The UI is already up an running when this code is called so it seems save to just call `Display.getDefault()` and not depend on PlatformUI.

Reply:
The `Platform` is only the OSGi framework (Equinox), but you most probably want that the Workbench is running (`PlatformUI`), as mentioned before you are using an E4 event here so that should be safe, but why then check for a running platform at all?

Reply:
You mean in runStartupCheck() the line ?
```
Platform.isRunning() && Platform.OS.isWindows() && !Platform.inDevelopmentMode()
```
I just added it as a extra safety net since it looks like that `Platform.OS.isWindows()` and `Platform.inDevelopmentMode()` will throw an exception if the platform is not running. Although I agree it should always run if that code is executed.

Reply:
Yes because if platform is not running there cant be no workbench, so no event.

On the other hand you assume that the Workbench is running and don't want to be safe there by not using `PlatformUI` ;-)
 if you want to be safe, you still need to use `UISynchronize` instead of Display ...

Reply:
> On the other hand you assume that the Workbench is running and don't want to be safe there by not using PlatformUI ;-)

Since I replaced the `getDisplay()` method by `Display.getDefault()` the only reference to `PlatformUI` is the call to `PlatformUI.isWorkbenchRunning()`, which is used to guard the creation of the link-text that can open the preference dialog. That should always be save.